dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorArm8.java

```diff
@@ -57,7 +57,7 @@ public List<Event> visitJoin(Join e) {
         return eventSequence(
                 load,
                 AArch64.DMB.newISHBarrier(),
-                newJumpUnless(new Atom(resultRegister, EQ, IValue.ZERO), e.getLabel())
+                newJumpUnless(new Atom(resultRegister, EQ, IValue.ZERO), (Label) e.getThread().getExit())
```

Comment:
shall we make `getExit()` directly return a label instead of always cast?

Reply:
I think that is reasonable. But do all threads have a label at the end? I'm thinking of litmus tests and init threads.

Reply:
Yes, see `ProgramBuilder:L52`

Reply:
Ok, but during construction at least, it is not true that `thread.getExit` is a label. `Thread.append` does not maintain that invariant either. I guess it would make sense to create the exit event on thread construction and have the `append` method insert events before the exit event (basically, the exit event always gets moved to the end).